### DISCUSSION TITLE
Blur after using pixmap

### DISCUSSION THREAD

**Discussion by Nurfen** [pymupdf/PyMuPDF#1116] [Q&A] [answered]
Hello JorjMcKie,

First of all congrats with creating the pymupdf package, i'm very impressed by its speed.

I'm trying to build a rescale function that rescales the pages in a pdf to A4 format.
However it always generates slightly blurred images, even when the pdf was already in A4 format.

Maybe it is a rounding error somewhere or I am using the wrong dimensions. 
Any idea?

```
import fitz
def pdf_rescale_A4(src:str,target:str):
    src = fitz.open(src)  # problem PDF
    doc = fitz.open()
    for ipage in src:
        if ipage.rect.width > ipage.rect.height:
            fmt = fitz.PaperRect("a4-l")  # landscape if input suggests
        else:
            fmt = fitz.PaperRect("a4")
        pix = ipage.get_pixmap()
        scale_w = fmt.width/pix.width
        scale_h = fmt.height/pix.height
        mat = fitz.Matrix(scale_w, scale_h)
        pix = ipage.get_pixmap(matrix=mat)
        page = doc.newPage(width = fmt.width, height = fmt.height)
        basestream = pix.pil_tobytes("JPEG")
        page.insert_image(page.rect, stream=basestream)
    doc.save(target)
```

Thanks in advance!

**Comment by JorjMcKie** [comment, marked as answer]
Make a higher quality image from the input page: Multiply the scale value with some factor > 1, eg take matrix * 2 instead. The aspect ratio will be kept, but you will have 4 times more pixels per page point in this case.

I'm not sure why you chose the term "rescale": you are aware that the output pages will consist of images only? If you look at them in a PDF reader and zoom, you will see pixeling at some magnification.

**Reply by Nurfen** [reply]
Thanks for the quick reply and the solution.
The image only format is wanted as a preprocessing step for our OCR. 

Below the updated code for completeness:

```
import fitz
def pdf_rescale_A4(src:str,target:str):
    src = fitz.open(src)  # problem PDF
    doc = fitz.open()
    for ipage in src:
        if ipage.rect.width > ipage.rect.height:
            fmt = fitz.PaperRect("a4-l")  # landscape if input suggests
        else:
            fmt = fitz.PaperRect("a4")
        pix = ipage.get_pixmap()
        scale_w = fmt.width/pix.width
        scale_h = fmt.height/pix.height
        mat = fitz.Matrix(scale_w * 2, scale_h * 2)
        pix = ipage.get_pixmap(matrix=mat)
        page = doc.newPage(width = fmt.width, height = fmt.height)
        basestream = pix.pil_tobytes("JPEG")
        page.insert_image(page.rect, stream=basestream)
    doc.save(target)
```
Have a nice day
